### PR TITLE
New Search API

### PR DESCRIPTION
**Pull request by DavidWChen**
## Description
- New search method
- Takes in a space separated query and finds all active mentors with matching strings in major, year, and name (finds all relevant mentors for each string, then intersects them)

- Random is still supported

Search by minor, courses, club, bio unsupported (not in master)

## Pls do 
- [x] Tested and working
- [x] Unit tests

## Test plan
Make test && Postman
image

### CODE REVIEW

File: src/users/tests.py [hunk 1]
```diff
@@ -214,17 +214,22 @@ class MentorsSearchTest(APITestCase):
...
+        self.major1 = factories.MajorFactory(name='Test_Major1')
+        self.profile1 = factories.ProfileFactory(year='1st', user=self.user1)
         self.mentor1 = factories.MentorFactory(major=self.major1, profile=self.profile1)
 
         self.major2 = factories.MajorFactory(name='Test_Major2')
         self.profile2 = factories.ProfileFactory(year='2nd')
         self.mentor2 = factories.MentorFactory(major=self.major2, profile=self.profile2)
 
+        self.major3 = factories.MajorFactory(name='Test_Major3')
+        self.profile3 = factories.ProfileFactory(year='2nd')
```
Comment (marktai) [inline src/users/tests.py]: I know this was already merged, but these really should be constants instead of '2nd'.

File: src/users/views.py [hunk 2]
```diff
@@ -228,26 +228,23 @@ class MentorsSearchView(generics.ListAPIView):
...
+            query = self.request.GET['query']
+            query = query.split(' ')
+
         
+            for item in query:
+                queryset = queryset.filter(
+                    Q(major__name__icontains = item) | 
+                    Q(profile__year__icontains = item) | 
+                    Q(profile__user__first_name__icontains = item) |
+                    Q(profile__user__last_name__icontains = item)
```
Comment (marktai) [inline src/users/views.py]: can we also translate `'first'` to `'1st'` and so on?  i'm a little worried about the odds of a user searching precisely "1st"

Comment (DavidWChen) [reply src/users/views.py]: As in if the original query contains 'first', add '1st' to the list of items iterated over?

**Comment by DavidWChen** [comment]
Is the way I did the "translation" for search fine? Direct translation seemed too rough, so I did a 
'''
if 'first' 
then 'first' or '1st'" 
'''
sort of implementation

**Comment by marktai** [comment]
yup looks good

**Pull request merged**